build.gradle

```diff
@@ -107,24 +139,46 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
-
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String checkForEapWithVersionHigherThan(String intellijMajorVersion) {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?)-EAP-SNAPSHOT.pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    String foundVersion = matcher.group(1)
+    if (!project.intellijVersions.eapOfLatestSupportedMajor) {
+      if (Integer.parseInt(project.majorPartOfBuildNumber(foundVersion)) > Integer.parseInt(intellijMajorVersion)) {
+        return foundVersion
+      }
+    } else {
+      if (project.fullIntellijVersionToNumber(foundVersion) > project.fullIntellijVersionToNumber(intellijMajorVersion)) {
```

Comment:
Hmmm won't `foundVersion` be of the format `222.1234.56`? and `intellijMajorVersion` of just `221`? Then `222123456 > 221` indeed... but I don't think that's the comparison we'd like to perform here!

Reply:
Oh, okay.... `intellijMajorVersion` will be of `222.789.12` format when we're enter this case... still, this logic is pretty convoluted... it must be simplified, lemme suggest something...

Reply:
but when intellijMajorVersion will be 221 (so taking from latestStable) we have first case, not this one

Reply:
Oh, okay.... `intellijMajorVersion` will be of `222.789.12` format when we're enter this case... still, this logic is pretty convoluted... it must be simplified, lemme suggest something:

Let's make sure this method does NOT depend on `project.intellijVersions` since it complicates the reasoning a lot. If needed, split into two separate method to be referenced from `String newerVersion = latestEapVersion ? ... : ...`

Reply:
> but when intellijMajorVersion will be 221 (so taking from latestStable) we have first case, not this one

Yeah I see now... but that's a silent assumption that's hard to figure out from code ☹️ let's rework the method to make it clear what format of version is expected where!